pkg/fleets/controller.go

```diff
@@ -444,7 +455,12 @@ func (c *Controller) rollingUpdateActive(fleet *agonesv1.Fleet, active *agonesv1
 	// make sure we don't end up with more than the configured max surge
 	maxSurge := surge + fleet.Spec.Replicas
 	replicas = fleet.UpperBoundReplicas(replicas + surge)
-	total := agonesv1.SumStatusReplicas(rest) + replicas
+	var total int32
+	if runtime.FeatureEnabled(runtime.FeatureRollingUpdateOnReady) {
+		total = agonesv1.SumSpecReplicas(rest) + replicas
```

Comment:
Can we add a comment here about why we're using the Spec, and not the status? Not sure I understand.

Reply:
well, this code was taken from kubernetes original source code. Let me add comments here.

Reply:
When we are counting all Replicas for ReplicaSet Kubernetes use `Spec.Replicas`:
https://github.com/kubernetes/kubernetes/blob/442a69c3bdf6fe8e525b05887e57d89db1e2f3a5/pkg/controller/deployment/rolling.go#L93
But in order to simplify this PR and solve original issue this change is not necessary. We can add it in Beta version of this new Feature.